server/app/db/crud/bots.py

```diff
@@ -0,0 +1,331 @@
+        bot_id = bot_type_data.id  # type: ignore
+    else:
+        bot_id = data.bot_type
+
+    # create the query
+    # "as" doesn't have proper support in pypika,
+    # so we have to write our own query
+    q = "SELECT bots.*, \
+        bot_types.id as bot_type_id,\
+        bot_types.name as bot_type_name \
+        from bots INNER JOIN bot_types ON bots.bot_type=bot_types.id\
+        WHERE "
+
+    if data.get_query_on() == "name":
+        q += "bots.name=%s"
+        log_query_with_arguments(q, [data.name])
+    else:
+        q += "bots.bot_type=%s"
+        log_query_with_arguments(q, [bot_id])
+
+    try:
+        if data.get_query_on() == "name":
+            await con.execute(q, [data.name])
+        else:
+            await con.execute(q, [bot_id])
+        rows = await con.fetchall()
+        resp = [bots_schema.create_botInDBInflated_from_tuple(row) for row in rows]
+        return resp, None
+    except Exception as e:
+        logging.error(
+            f"Couldn't query bots (inflated) with {data.dict()} with query={q}",
+            f" and args={[data.name, bot_id]} due to {e}",
+        )
+        return [], None
+
+
+async def _get_bots_with_details_uninflated(
+    con: Cursor, name: str = "", bot_type: Union[str, int] = 0
+) -> Tuple[List[bots_schema.BotInDB], Union[Exception, None]]:
+    """Function to query uninflated details"""
+
+    logging.info(f"Quering bots with {name=} and {bot_type=} uninflated")
+
+    # validating data
+    try:
+        data = bots_schema.QueryBot(name=name, bot_type=bot_type)
+    except ValidationError as e:
+        # validation failed, return errors
+        logging.error(
+            f"Couldn't to query bots with {name=} and {bot_type=}",
+            f" as validation failed with the error(s): {e.errors()}",
+        )
+        return [], e
+
+    bot_id = 0
+    if type(data.bot_type) is str:
+        # name of the bot_type is given, query for the bot_type_id and
+        # use the to query the bot data
+        bot_type_data, err = await get_bot_type_with_given_name(con, data.bot_type)
+        if err:
+            return [], err
+        bot_id = bot_type_data.id  # type: ignore
+    else:
+        bot_id = data.bot_type
+
+    criteria: List[Any] = []
+    if data.get_query_on() == "name":
+        criteria.append(bots.name == data.name)
+    else:
+        criteria.append(bots.bot_id == bot_id)
+
+    # creating query
+    # selecting * -> criteria contraint
+    q = MySQLQuery.from_("bots").select("*").Where(Criterion.all(criteria))
+
+    # executing the query
+    try:
+        await con.execute(q)
+        rows = await con.fetchall()
+        resp = [bots_schema.create_BotInDB_from_tuple(row) for row in rows]
+        return resp, None
+    except Exception as e:
+        logging.error(
+            "Couldn't query bots(uninflated) with"
+            + f"{data.dict()} with query={q} due to {e}",
+        )
+        return [], None
+
+
+async def update_bot_name_with_id(
```

Comment:
I think we also need a function to change the current type of bot to some other type of bot. For eg, Change a particular News Bot to ~Slave~ Bot

Reply:
lol okay